Mock server profile, getAuthenticatedUser, and graphql schema updates

andersan opened the pull request:
This PR introduces 4 minor changes:
1. Move profile completion logic to the User in the mock backend
2. ~~Don't return `null` for `getAuthenticatedUser` when the user is logged out. Do we want the mock backend to throw an error in this case in order to match the mmdata backend functionality? See [this trello task](https://trello.com/c/KeSnySPu/49-servererror-when-launching-app-without-being-logged-in) for more context.~~ no longer needed
3. Bulk updates to graphql schema, copied in directly from the Apollo Server ui.
4. ~~In the `myInbox` query, return channel messages with a channelId stored in `channels` in the `state.ts` file. I don't know if this is really necessary.~~ this was implemented in #46

ashlilili commented on mm-mock-server/schema.graphql:
Seems like the profileCompletionPercentage field under User added in https://github.com/micromentor-team/mm-flutter-app/pull/34 was removed - this would cause the app to fail because the api kGetUserProfileInfo , which is being used for reminder banner, assumes this field profileCompletionPercentage under User.

andersan replied:
Good catch, Ashley. I pulled these graphql updates from a remote apollo server rather than my latest local. This is fixed in the latest commit on this PR.

ashlilili commented on mm-mock-server/src/mocks/queries.ts:
```diff
@@ -22,20 +22,10 @@ export function mockQueries(serverState: MockServerState) {
-            }
             return serverState.loggedInUser;
         },
-        getUserProfileInfo: () => {
```
If we remove getUserProfileInfo, we should change kGetUserProfileInfo under lib/data/models/user/user_api.dart to querying getAuthenticatedUser?

andersan replied:
Yes, made this change in the latest updates

ashlilili reviewed (commented):
Thanks for catching this again and cleaning up the backend changes!

m-rosario commented on mm-mock-server/src/mocks/queries.ts:
```diff
@@ -22,20 +22,10 @@ export function mockQueries(serverState: MockServerState) {
             return serverState.otherUsers.concat([serverState.loggedInUser]);
         },
         getAuthenticatedUser: () => {
-            if (!serverState.loggedIn) {
-                return null;
-            }
```
I believe the real backend returns null when the user is not logged in, aside from throwing an exception. From what Holger mentioned I believe this means that now we can always expect a User and we will instead verify some other field (or more specific exception) to determine if the user is authorized.

andersan replied:
You're right, didn't realize that the backend was returning `data: null` on errors.

```json
{"errors":[{"message":"Cannot return null for non-nullable field Query.getAuthenticatedUser.","locations":[{"line":2,"column":3}],"path":["getAuthenticatedUser"],"extensions":{"code":"INTERNAL_SERVER_ERROR","stacktrace":["Error: Cannot return null for non-nullable field Query.getAuthenticatedUser.","    at completeValue (/Users/chris/Documents/mm3/mmdata/node_modules/graphql/execution/execute.js:594:13)","    at /Users/chris/Documents/mm3/mmdata/node_modules/graphql/execution/execute.js:486:9","    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","    at async Promise.all (index 0)","    at async execute (file:///Users/chris/Documents/mm3/mmdata/node_modules/@apollo/server/dist/esm/requestPipeline.js:220:37)","    at async processGraphQLRequest (file:///Users/chris/Documents/mm3/mmdata/node_modules/@apollo/server/dist/esm/requestPipeline.js:155:32)","    at async internalExecuteOperation (file:///Users/chris/Documents/mm3/mmdata/node_modules/@apollo/server/dist/esm/ApolloServer.js:585:16)","    at async runHttpQuery (file:///Users/chris/Documents/mm3/mmdata/node_modules/@apollo/server/dist/esm/runHttpQuery.js:129:29)","    at async runPotentiallyBatchedHttpQuery (file:///Users/chris/Documents/mm3/mmdata/node_modules/@apollo/server/dist/esm/httpBatching.js:34:16)","    at async ApolloServer.executeHTTPGraphQLRequest (file:///Users/chris/Documents/mm3/mmdata/node_modules/@apollo/server/dist/esm/ApolloServer.js:496:20)"]}}],"data":null}
```

andersan commented:
Will fix dart formatting locally in the morning and then this should be ready for another review. I added `updatedAt` to the `getAuthenticatedUser` response and added that to the User provider Flutter reminder widget. Flutter is building for me locally and I can run the app in chrome with these changes.

ashlilili reviewed (dismissed):
Thanks Chris!

andersan commented:
@m-rosario could you review this again when you have a chance? I rebased with main after the changes over the past few days.

Thanks!

andersan commented:
The tests were appearing as pending when I tried to merge this. There was some bug where I had to refresh this PR page to see the results which I was already seeing [in Github actions](https://github.com/micromentor-team/mm-flutter-app/actions/runs/5487635611).
